build-tools/packages/build-cli/src/commands/modify/fluid-imports.ts

```diff
@@ -292,24 +375,89 @@ async function updateImports(
+ * Parses an import declaration for processing as a Fluid Framework basic import.
+ * Non-FF and complex imports are ignored (returns undefined).
  *
  * @param importDeclaration - the import declaration to check.
- * @returns a tuple of `[module specifier, subpath]`
+ * @param index - the current index of import block array.
+ * @param log - logger.
+ * @returns a {@link FluidImportDataPresent} metadata object
  */
-function parseImport(importDeclaration: ImportDeclaration): [string, string] {
+function parseImport(
+	importDeclaration: ImportDeclaration,
+	index: number,
+	log: CommandLogger,
+): FluidImportDataPresent | undefined {
 	const moduleSpecifier = importDeclaration.getModuleSpecifierValue();
 	const modulePieces = moduleSpecifier.split("/");
-	const moduleName = modulePieces.slice(0, 2).join("/");
-	const subpath = modulePieces.length === 3 ? modulePieces[2] : "public";
-	return [moduleName, subpath];
+	const levelIndex = moduleSpecifier.startsWith("@") ? 2 : 1;
+	const packageName = modulePieces.slice(0, levelIndex).join("/");
+	const level = modulePieces.length > levelIndex ? modulePieces[levelIndex] : "public";
+	if (!isKnownLevel(level)) {
+		return undefined;
+	}
+	// Check for complicated path - beyond basic leveled import
+	if (modulePieces.length > levelIndex + 1) {
+		return undefined;
+	}
+	// Check for Fluid import
+	if (!isFluidImport(packageName)) {
+		return undefined;
+	}
+	// Check namespace imports which are checked trivially for API level use.
+	if (importDeclaration.getNamespaceImport() !== undefined) {
+		log.verbose(`\tSkipping namespace import of ${moduleSpecifier}`);
+		return undefined;
+	}
+
+	const order = knownLevels.indexOf(level) * 2 + (importDeclaration.isTypeOnly() ? 0 : 1);
+	return {
+		importDeclaration,
+		declaration: {
+			isTypeOnly: importDeclaration.isTypeOnly(),
+			moduleSpecifier,
+			namedImports: [],
+		},
+		index,
+		packageName,
+		level,
+		order,
+		originallyUnassigned: isImportUnassigned(importDeclaration),
+	};
 }
 
-function isFluidImport(importDeclaration: ImportDeclaration): boolean {
-	const moduleSpecifier = importDeclaration.getModuleSpecifierValue();
+/**
+ * Parses a source file for basic static Fluid Framework imports.
+ * Non-FF and complex imports are excluded.
+ *
+ * @param sourceFile - the ${@link SourceFile} to parse.
+ * @param log - logger.
+ * @returns an array of {@link FluidImportDataPresent} metadata objects
+ */
+function parseFluidImports(
+	sourceFile: SourceFile,
+	log: CommandLogger,
+): FluidImportDataPresent[] {
+	return sourceFile
+		.getImportDeclarations()
+		.map((importDecl, index) => parseImport(importDecl, index, log))
+		.filter(
+			(v) => v !== undefined,
+		) /* no undefined elements remain */ as FluidImportDataPresent[];
+}
+
+function isFluidImport(packageName: string): boolean {
+	return (
+		packageName.startsWith("@fluid") ||
+		["fluid-framework", "tinylicious"].includes(packageName)
```

Comment:
Might be worth pulling these into a const earlier in the file. Could also consider getting the list from the fluidBuild config (`policy.packageNames.unscopedPackages`). Not blocking - I wrote this code. :)

Reply:
Is there a default config? We want to use this outside of FF we should have the right default set.
I pulled out to a const and unicorn says use a Set instead of array.

Reply:
Ahhh, you're right about default config. It's empty. Never mind.